### ISSUE TITLE
Backspace results in unintentional annotation deletion which also re-appears after re-opening PDF

### ISSUE THREAD

**Issue by mrtcode** [zotero/zotero#2292]
https://forums.zotero.org/discussion/93648/too-many-backspaces-will-hide-annotation

**Comment by mrtcode** [comment]
The current behaviour is that if annotation comment is empty, backspace (on key up) deletes selected annotation no matter if you just selected the annotation, or you were deleting comment text and pressing backspace too many times. In the later case the deletion is unexpected. But this case doesn't seem to be common in practice. Maybe having undo functionally would make this issue obsolete. Otherwise we need to track if user was editing comment and prevent annotation deletion until it's re-selected.

**Comment by dstillman** [comment]
No, I don't think Undo is relevant here — you shouldn't be able to delete the annotation just by continuing to press Backspace in the comment field. The comment field should stay active, capturing Backspace as a no-op, until you move focus elsewhere.

**Comment by dstillman** [comment]
I actually can't reliably reproduce this. Most of the time it behaves correctly, with the comment field staying open and Backspace not having any effect. But sometimes — maybe if you create the annotation and start typing quickly enough, or some other sort of race condition — Backspace deletes the annotation.

**Comment by mrtcode** [comment]
This happens when the annotation (or anything in its widget including comment text box) is clicked, and the flag that prevents deletion with backspace gets cleared.

**Comment by dstillman** [comment]
Ah, yeah, that's it — type a letter, click again in the comment box, and press Backspace twice.

**Comment by UdeRecife** [comment]
Is there a way to disable this 'backspace' behavior? I don't want it to delete anything other than the text I type. Thanks in advance.
